File: cpp_solutions/2511. Maximum Enemy Forts That Can Be Captured.cpp
class Solution {
public:
    int captureForts(vector<int>& forts) {

        int ans = 0;
        int j = 0;
        for(int i=0; i<forts.size();++i){
            if (forts[i]!=0){
                if(forts[j] == -forts[i]){
                    ans = max(ans, i-j-1);
                }
                j = i;
            }
        }
        return ans;
    }
};

File: python3_solution_set2/2511. Maximum Enemy Forts That Can Be Captured.py
class Solution:
    def captureForts(self, forts: List[int]) -> int:

        ## -1, 0 , 1
        ## no, enemy, my

        j = 0
        ans = 0
        for i in range(len(forts)):
            if forts[i] != 0:
                if forts[j] == -forts[i]:
                    ans = max(ans, i-j-1)
                j = i
        return ans


